internal/controller/controllers/cluster_controller.go

```diff
@@ -79,19 +81,56 @@ func (r *ClusterReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
 		return r.updateState(ctx, cluster, nil, err)
 	}
 
-	// check for updates from user, compare spec and update if needed
-	updated, result, err := r.updateIfNeeded(ctx, cluster, c)
-	if err != nil {
-		return r.updateState(ctx, cluster, c, err)
+	if r.canClusterBeUpdated(c) {
+		var updated bool
+		var result ctrl.Result
+		// check for updates from user, compare spec and update if needed
+		updated, result, err = r.updateIfNeeded(ctx, cluster, c)
+		if err != nil {
+			return r.updateState(ctx, cluster, c, err)
+		}
+
+		if updated {
+			return result, err
+		}
 	}
 
-	if updated {
-		return result, err
+	if r.isReadyForInstallation(cluster, c) {
+		var ic *common.Cluster
+		ic, err = r.Installer.InstallClusterInternal(ctx, installer.InstallClusterParams{
+			ClusterID: *c.ID,
+		})
+		if err != nil {
+			return r.updateState(ctx, cluster, c, err)
+		}
+		return r.updateState(ctx, cluster, ic, nil)
 	}
 
 	return r.updateState(ctx, cluster, c, nil)
 }
 
+func (r *ClusterReconciler) canClusterBeUpdated(c *common.Cluster) bool {
+	readyForUpdateStates := []string{models.ClusterStatusReady, models.ClusterStatusPendingForInput,
+		models.ClusterStatusInsufficient}
+	return funk.ContainsString(readyForUpdateStates, swag.StringValue(c.Status))
+}
+
+func (r *ClusterReconciler) isReadyForInstallation(cluster *adiiov1alpha1.Cluster, c *common.Cluster) bool {
+	if swag.StringValue(c.Status) != models.ClusterStatusReady {
```

Comment:
Perhaps we can reuse [IsReadyForInstallation()](
https://github.com/openshift/assisted-service/blob/432f69c82c537cc159e7a7379f398b0b26c9298f/internal/cluster/cluster.go#L639 ) here